bryjbrown/islandora_scholar
# #280 fix to be integrated via #285 (ISLANDORA-1923)

**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-1923

* #285
* #280

# What does this Pull Request do?

Solves an issue in #280, makes small improvements and streamlines #285 with #280.

# What's new?

* Make checkbox "Use legacy retrieval mode" behave as expected
* Improve readability of `islandora_doi_use_legacy_mode()`
* Ensure all variables are deleted upon uninstall
* Default to https when querying CrossRef
* Change filename case of doi_populator's readme
* Use constants from #280 instead of literals (where appropriate)
* Restrict PID validity check to legacy-mode
* Prefer "identifier" over "email"

# How should this be tested?

* Check if (de-)selection of "Use legacy retrieval mode" gets stored under all circumstances (PID defined/undefined/valid/invalid before PR from 7.x, as well as afterwards - it is probably easier to play with `variable_del()` on `/devel/php`)
* Uninstall the module and check if the four `islandora_doi_...` variables are deleted
* Check if queries in legacy mode are going over https (e.g., run `sudo tcpdump -Q out dst host www.crossref.org` on your VM while querying CrossRef)
* Check if PID validity check only happens when "Use legacy retrieval mode" is de-selected

# Additional Notes:

* It is necessary to run `drush updb` after this PR (in order to update the default for OpenURL to use https).
* @bryjbrown It was not my intention to take the integration of #280 into #285 out of your hands. What you see here is merely a suggestion. Feel free to revert back one commit (or even two, if you feel that the small improvements should be addressed via a separate JIRA-ticket - I was just thinking to hurry things a bit before the 7.x-1.11 code freeze)!


# Interested parties
@bryjbrown
